.github/workflows/ci.yml

```
@@ -307,8 +307,116 @@ jobs:
 #          result-path: ${{matrix.report.result-path || './reports'}}
 #          meta-path: ${{matrix.report.meta-path || './reports'}}
 
+
+  python-build:
+    needs: [setup]
+    if: ${{fromJSON(needs.setup.outputs.builds).python}}
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{fromJSON(needs.setup.outputs.builds).python}}
+    name: Build / ${{matrix.display-name || 'python (skipped)'}}
+    timeout-minutes: 20
+    runs-on: ${{matrix.runner || 'ubuntu-20.04'}} # there are no old pythons for new Ubuntu
+    container: ${{matrix.container}}
+    env: ${{matrix.env}}
+    steps:
+      - name: Check-out repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{matrix.python-version}}
+      - name: Restore build artifacts
+        if: ${{matrix.builds}}
```

it is always an array so it never be falsy
```suggestion
        if: ${{join(matrix.builds)}}
```